### PR TITLE
[WIP] [ESPnet2] Mask-CTC

### PR DESCRIPTION
**Pull request by YosukeHiguchi**
This PR adds [Mask-CTC](https://arxiv.org/abs/2005.08700)-based non-autoregressive ASR model to ESPnet2.

### CODE REVIEW

**Comment by YosukeHiguchi** [comment]
@sw005320 Are these CI errors related to my commits?

**Comment by YosukeHiguchi** [comment]
I think the implementation is ready. I'll also add WSJ results along with baseline CTC results.

File: egs2/wsj/asr1/conf/tuning/train_asr_transformer_maskctc.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+batch_type: folded
```
Comment (sw005320) [inline egs2/wsj/asr1/conf/tuning/train_asr_transformer_maskctc.yaml:1]: Can you add some remarks (comments) on which parts are different from the original training config?
Obviously decoder, right? Other parts?

File: espnet2/tasks/asr.py [hunk 2]
```diff
@@ -232,12 +247,6 @@ def add_task_arguments(cls, parser: argparse.ArgumentParser):
             default=None,
             help="The keyword arguments for joint network class.",
         )
-        group.add_argument(
-            "--model_conf",
```
Comment (sw005320) [inline espnet2/tasks/asr.py:236]: Sorry I cannot understand this part.
Why this option is removed?

**Review by sw005320 — COMMENTED** [review]
LGTM.
Some minor comments

File: espnet2/tasks/asr.py (hunk 2 above)
Comment (YosukeHiguchi) [reply espnet2/tasks/asr.py:236]: I've defined `model_choices` in L112 and added it to `class_choices_list` in L187. With this change, arguments from --model and --model_conf should be loaded by the `add_arguments` method in L330.

Comment (sw005320) [reply espnet2/tasks/asr.py:236]: Thanks. How is the help message (`help="The keyword arguments for model class."`) handled?

Comment (YosukeHiguchi) [reply espnet2/tasks/asr.py:236]: It will be handled in https://github.com/espnet/espnet/blob/master/espnet2/train/class_choices.py#L94

Comment (sw005320) [reply espnet2/tasks/asr.py:236]: thanks!

**Comment by sw005320** [comment]
Thanks a lot.
It would be great if you add some results, models, and comments to the config files.

**Pull request merged**